composables/useSingleReport.js

```diff
@@ -1,35 +1,73 @@
-import { useQuery } from "vue-query";
-import { usePackageState } from "./usePackagesState";
-import { watch } from "vue";
+import { useQuery, useMutation, useQueryClient } from "@tanstack/vue-query";
+
 const fetcher = (url) =>
   $fetch(url, {
     headers: useRequestHeaders(["cookie"]),
   });
 
 export const useSingleReport = () => {
   const route = useRoute();
-  const { setState } = usePackageState();
-
+  const queryClient = useQueryClient();
   const {
-    data: report,
-    error: fetchError,
     isLoading,
-  } = useQuery(
-    [`report`, route.params.id],
-    () => fetcher(`/api/reports/${route.params.id}`),
-    {
-      staleTime: 5000,
-      retry: 0,
-    }
-  );
-
-  watch(report, (newReport) => {
-    if (newReport) setState(newReport.state);
+    data: report,
+    isError,
+  } = useQuery({
+    queryKey: [`report`, route.params.id],
+    queryFn: () => fetcher(`/api/reports/${route.params.id}`),
+    retry: 0,
+  });
+
+  const areAllEulasApproved = ({ name, suggested_criteria }) => {
+    return suggested_criteria.every(
+      (criteria) => report.value?.state?.[name]?.[criteria]
+    );
+  };
+
+  const { mutateAsync: mutateApproval } = useMutation({
+    mutationFn: ({ pkg, eula }) =>
+      $fetch(`/api/reports/${route.params.id}`, {
+        method: "PUT",
+        body: {
+          ...report.value,
+          [pkg]: {
+            ...report.value?.state?.[pkg],
+            [eula]: report.value?.state?.[pkg]?.[eula] ? false : true,
+          },
+        },
+      }),
+    onSuccess: () => {
```

Comment:
**question:** Are we handling these errors?

**suggestion, non-blocking, but followup needed:** Let's add some error handling.

Reply:
Wasn't in the aproval toggle, fixed